tests/unit/test_basic_functionality.py

```diff
@@ -648,10 +676,51 @@ def test_retrieval_behavior() -> None:
         lct = LayeredConfigTree(layers=layer_list)
         for layer in scenario:
             lct.update({default_cfg_value: layer}, layer=layer)
-        assert lct.get_from_layer(default_cfg_value) == layer_outer
-        assert lct.get_from_layer(default_cfg_value, layer=layer_outer) == layer_outer
-        assert lct.get_from_layer(default_cfg_value, layer=layer_middle) == layer_middle
-        assert lct.get_from_layer(default_cfg_value, layer=layer_inner) == layer_inner
+        assert lct.get(default_cfg_value) == layer_outer
+        assert lct.get(default_cfg_value, layer=layer_outer) == layer_outer
+        assert lct.get(default_cfg_value, layer=layer_middle) == layer_middle
+        assert lct.get(default_cfg_value, layer=layer_inner) == layer_inner
+
+
+def test_nested_retrieval_from_layer() -> None:
+    lct = LayeredConfigTree(layers=["base", "override"])
+    lct.update(
+        {
+            "outer": {
+                "inner": {
+                    "one": 1,
+                    "two": 2,
+                },
+            },
+        },
+        layer="base",
+    )
+    lct.update(
+        {
+            "outer": {
+                "inner": {
+                    "one": 100,
+                    "two": 200,
+                },
+                "new": "foo",
+            },
+        },
+        layer="override",
+    )
+    assert (
+        lct.get(["outer", "inner", "one"])
+        == lct.get(["outer", "inner", "one"], layer="override")
+        == 100
+    )
+    assert (
+        lct.get(["outer", "inner", "two"])
+        == lct.get(["outer", "inner", "two"], layer="override")
+        == 200
+    )
+    assert lct.get(["outer", "new"]) == lct.get(["outer", "new"], layer="override") == "foo"
+    assert lct.get(["outer", "oops"], "missing") == "missing"
+    assert lct.get(["outer", "inner", "one"], layer="base") == 1
+    assert lct.get(["outer", "inner", "two"], layer="base") == 2
```

Comment:
Shouldn't we have a case where both `default_value` and `layer` are provided?

Reply:
I split this test up a bit and added this